lib/Service/DocumentService.php

```diff
@@ -490,4 +496,41 @@ private function ensureDocumentsFolder(): bool {
 
 		return true;
 	}
+
+	public function lock(int $fileId): bool {
+		if (!$this->lockManager->isLockProviderAvailable()) {
+			return true;
+		}
+
+		$file = $this->getFileById($fileId);
+		try {
+			$this->lockManager->lock(new LockContext(
+				$file,
+				ILock::TYPE_APP,
+				Application::APP_NAME
+			));
+		} catch (NoLockProviderException $e) {
+		} catch (OwnerLockedException $e) {
+			return false;
+		} catch (PreConditionNotMetException $e) {
+		}
+		return true;
+	}
+
+	public function unlock(int $fileId): void {
+		if (!$this->lockManager->isLockProviderAvailable()) {
+			return;
+		}
+
+		$file = $this->getFileById($fileId);
+		try {
+			$this->lockManager->unlock(new LockContext(
+				$file,
+				ILock::TYPE_APP,
+				Application::APP_NAME
+			));
+		} catch (NoLockProviderException $e) {
+		} catch (PreConditionNotMetException $e) {
```

Comment:
No warning logs here?

Reply:
I'd let the actual lock provider handle that, from the text app side there is no special interest if we unlock a file that has already been unlocked.